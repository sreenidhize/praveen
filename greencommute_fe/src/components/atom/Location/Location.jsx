import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles(() => ({
  label: {
    fontSize: "16px",
    paddingLeft: "10px",
    display: "inline",
    position: "absolute",
  },
}));

const Location = (props) => {
  const { label, icon } = props;
  const classes = useStyles();

  return (
    <div>
      {icon}
      <Typography className={classes.label} variant="h4">
        {label}
      </Typography>
    </div>
  );
};

Location.propTypes = {
  label: PropTypes.string,
  icon: PropTypes.any,
};

Location.defaultProps = {
  label: "label",
};

export default Location;
