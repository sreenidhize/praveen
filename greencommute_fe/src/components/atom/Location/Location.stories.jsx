import React from "react";
import Location from "./Location";
import LocationOnOutlinedIcon from "@material-ui/icons/LocationOnOutlined";

export default {
  title: "Molecules/Location",
  component: Location,
};
const LocationTemplate = (args) => <Location {...args} />;

export const Primary = LocationTemplate.bind({});

Primary.args = {
  label: "East Maredupally, Hyderabad, Telangana",
  icon: <LocationOnOutlinedIcon color="disabled" />,
};
