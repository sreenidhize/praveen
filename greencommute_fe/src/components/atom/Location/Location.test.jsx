import React from "react";
import Location from "./Location";
import { render, screen } from "@testing-library/react";
import LocationOnOutlinedIcon from "@material-ui/icons/LocationOnOutlined";

describe("Location renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<Location />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Location should not be empty", () => {
    const { container } = render(
      <Location
        label="East Maredupally,Hyderabad,Telangana"
        icon={<LocationOnOutlinedIcon color="disabled" />}
      />
    );
    expect(container.getElementsByTagName("h4").length).toBe(1);
  });
});
