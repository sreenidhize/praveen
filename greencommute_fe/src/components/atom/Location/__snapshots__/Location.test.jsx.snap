// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h4
      class="MuiTypography-root makeStyles-label-1 MuiTypography-h4"
    >
      label
    </h4>
  </div>
</DocumentFragment>
`;
